app/tests/content/test_badge_integration.py

```diff
@@ -0,0 +1,160 @@
+from datetime import timedelta
+
+from rest_framework import status
+
+import pytest
+
+from app.content.factories.badge_factory import BadgeFactory
+from app.util.test_utils import get_api_client
+from app.util.utils import now
+
+API_BADGE_BASE_URL = "/badges/"
+
+
+def _get_badge_leaderboard_url():
+    return f"{API_BADGE_BASE_URL}leaderboard/"
+
+
+def _get_badge_specific_leaderboard_url(badge):
+    return f"{API_BADGE_BASE_URL}{badge.id}/leaderboard/"
+
+
+def _get_user_badge_url():
+    return f"/users/me{API_BADGE_BASE_URL}"
+
+
+def _get_badge_flag(badge):
+    return {"flag": badge.flag}
+
+
+@pytest.mark.django_db
+def test_get_badge_leaderboard_as_anonymous_user(default_client):
+    """An anonymous user should not be able to see the leaderboard"""
+
+    url = _get_badge_leaderboard_url()
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_get_badge_spesific_leaderboard_as_anonymous_user(default_client, badge):
+    """An anonymous user should not be able to see badge-specific leaderboard"""
+
+    url = _get_badge_specific_leaderboard_url(badge)
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_get_request_for_user_badges_as_anonymous_user(default_client):
+    """An anonymous user should not be able to do a get request for a user badge"""
+
+    url = _get_user_badge_url()
+    response = default_client.get(url)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
Kan være lurt å ha samme tester for medlemmer med da forventet status_kode == 200